packages/blueprints/gen-ai-chatbot/src/blueprint.ts

```diff
@@ -270,16 +301,41 @@ export class Blueprint extends ParentBlueprint {
     }
   }
 
+  /**
+   * Checks if this is running in resynthesis mode and the model of Claude used by the blueprint is
+   * being updated. The version of this blueprint that uses Claude 2 is not compatible with Claude 3
+   * and would result in the customer losing conversation history.
+   */
+  private claudeVersionCheck() {
+    // Instantations before or on this version use Claude 2
+    const claude2MaxBlueprintVersion = '0.3.86';
+    const currentInstantiation =
+      this.context.project?.blueprint?.instantiationId &&
+      this.context.project.blueprint.instantiations?.find(i => i.id === this.context.project.blueprint.instantiationId);
+    return currentInstantiation
+      ? {
+        from2To3: semver.lte(currentInstantiation.versionId, claude2MaxBlueprintVersion),
+        from3To2: semver.lt(claude2MaxBlueprintVersion, currentInstantiation.versionId),
+      }
+      : undefined;
+  }
+
   private seedRepository(repository: SourceRepository, options: Options) {
     repository.copyStaticFiles({
```

Comment:
Do you need to set resynth strats to anything in particular as part of this update?